File: controllers/control.js
const axios = require('axios');
const ConfigDao = require('./../daos/configDao.js');


let msServerPort;
let msServerIp;
let msServerProtocol;
let url; 


async function initializeConfig() {

    const configDao = new ConfigDao();
    msServerPort = await configDao.getValueByKey('msServerPort');
    msServerIp = await configDao.getValueByKey('msServerIp');
    msServerProtocol = await configDao.getValueByKey('msServerProtocol');
}

 



const getStations = async () => {
    
    await initializeConfig();
    const LINE_NUMBER = 2; 
    const url = msServerProtocol + '://' + msServerIp + ":" + msServerPort;
   
    const stationUrl = url + '/infra/line/' + LINE_NUMBER;
    console.log(stationUrl);

    try {

        const response = await axios.get(stationUrl);
        console.log('Data fetched successfully:', response.data);

    } catch (error) {

        console.error('Error fetching data:', error.message);
    }
}


const getInterStations = async () => {

    const LINE_NUMBER = 2; 
  
    const interstationUrl = url +   '/infra/interstation/line/' + LINE_NUMBER ;

    try {

        const response = await axios.get(interstationUrl);
        console.log('Data fetched successfully:', response.data);

    } catch (error) {
        console.error('Error fetching data:', error.message);
    }
}

getStations();
getInterStations();

module.exports = { getStations, getInterStations };

File: controllers/control.py
import requests
from daos.ConfigDao import ConfigDao 
from entities.fliterApi import *

def initialize_config():
    
    config_dao = ConfigDao()
    ms_server_port = config_dao.get_value_by_key('msServerPort')
    ms_server_ip = config_dao.get_value_by_key('msServerIp')
    ms_server_protocol = config_dao.get_value_by_key('msServerProtocol')

    url = f"{ms_server_protocol}://{ms_server_ip}:{ms_server_port}"
    return url
def get_stations():
    
    url = initialize_config()
    line_number = 2
    station_url = f"{url}/infra/line/{line_number}"

    try:
        response = requests.get(station_url)
        response.raise_for_status()  # Raise error for bad status codes
        data = response.json()
        return data         
    except requests.exceptions.RequestException as e:
            print('Error fetching data:', str(e))
        

def get_interstations():
    url = initialize_config()
    line_number = 2
    interstation_url = f"{url}/infra/interstation/line/{line_number}"

    try:
        response = requests.get(interstation_url)
        response.raise_for_status()  # Raise error for bad status codes
        data = response.json()
        return data         
    except requests.exceptions.RequestException as e:
        print('Error fetching data:', str(e))

def main():
    
    stations = get_stations()
    interstations = get_interstations()

   
    return stations , interstations 

if __name__ == "__main__":
    main()